### PR TITLE
Description v4

### PR DESCRIPTION
**Pull request by arcadiafalcone**
## Why was this change made?
Updates openapi spec to match COCINA description schema v4.

### CODE REVIEW

File: lib/cocina/models/descriptive_admin_metadata.rb [hunk 1]
```diff
@@ -5,8 +5,10 @@ module Models
     class DescriptiveAdminMetadata < Struct
       attribute :contributor, Types::Strict::Array.of(Contributor).meta(omittable: true)
       attribute :event, Types::Strict::Array.of(Event).meta(omittable: true)
-      attribute :language, Types::Strict::Array.of(DescriptiveValue).meta(omittable: true)
+      attribute :language, Types::Strict::Array.of(Language).meta(omittable: true)
```
Comment (ndushay) [inline lib/cocina/models/descriptive_admin_metadata.rb:8]: not backwards compatible - be prepared for problematic data?

File: lib/cocina/models/descriptive_basic_value.rb [hunk 2]
```diff
@@ -13,9 +13,14 @@ class DescriptiveBasicValue < Struct
       attribute :code, Types::Strict::String.meta(omittable: true)
       # URI value of the descriptive element.
       attribute :uri, Types::Strict::String.meta(omittable: true)
-      attribute :standard, Types::Strict::Array.of(Types::Strict::String).meta(omittable: true)
-      attribute :encoding, Types::Strict::Array.of(Types::Strict::String).meta(omittable: true)
+      attribute :standard, Standard.optional.meta(omittable: true)
+      attribute :encoding, Standard.optional.meta(omittable: true)
```
Comment (ndushay) [inline lib/cocina/models/descriptive_basic_value.rb:17]: not backwards compatible - be prepared for problematic data?

File: lib/cocina/models/descriptive_structured_value.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@
 module Cocina
   module Models
     class DescriptiveStructuredValue < Struct
-      attribute :structuredValue, Types::Strict::Array.of(DescriptiveBasicValue).meta(omittable: true)
+      attribute :structuredValue, Types::Strict::Array.of(DescriptiveValue).meta(omittable: true)
```
Comment (ndushay) [inline lib/cocina/models/descriptive_structured_value.rb:6]: not backwards compatible - be prepared for problematic data?

File: lib/cocina/models/descriptive_value.rb [hunk 4]
```diff
@@ -13,10 +13,14 @@ class DescriptiveValue < Struct
       attribute :code, Types::Strict::String.meta(omittable: true)
       # URI value of the descriptive element.
       attribute :uri, Types::Strict::String.meta(omittable: true)
-      attribute :standard, Types::Strict::Array.of(Types::Strict::String).meta(omittable: true)
-      attribute :encoding, Types::Strict::Array.of(Types::Strict::String).meta(omittable: true)
+      attribute :standard, Standard.optional.meta(omittable: true)
+      attribute :encoding, Standard.optional.meta(omittable: true)
```
Comment (ndushay) [inline lib/cocina/models/descriptive_value.rb:17]: not backwards compatible - be prepared for problematic data?

File: lib/cocina/models/descriptive_value_required.rb [hunk 5]
```diff
@@ -13,10 +13,14 @@ class DescriptiveValueRequired < Struct
       attribute :code, Types::Strict::String.meta(omittable: true)
       # URI value of the descriptive element.
       attribute :uri, Types::Strict::String.meta(omittable: true)
-      attribute :standard, Types::Strict::Array.of(Types::Strict::String).meta(omittable: true)
-      attribute :encoding, Types::Strict::Array.of(Types::Strict::String).meta(omittable: true)
+      attribute :standard, Standard.optional.meta(omittable: true)
+      attribute :encoding, Standard.optional.meta(omittable: true)
```
Comment (ndushay) [inline lib/cocina/models/descriptive_value_required.rb:17]: not backwards compatible - be prepared for problematic data?

File: spec/cocina/generator/schema_array_spec.rb [hunk 6]
```diff
@@ -5,14 +5,14 @@
...
-    end
-  end
+  # context 'when an array of datatypes' do
+  #   # DescriptiveBasicValue.standard is an array of strings
+  #   let(:value) { Cocina::Models::DescriptiveBasicValue.new(standard: %w[marc bibframe]) }
+  # 
+  #   it 'maps datatypes' do
+  #     expect(value.standard).to eq(%w[marc bibframe])
+  #   end
+  # end
```
Comment (ndushay) [inline spec/cocina/generator/schema_array_spec.rb:15]: should this be updated to work with new types?

File: lib/cocina/models/description.rb [hunk 7]
```diff
@@ -7,12 +7,14 @@ class Description < Struct
       attribute :contributor, Types::Strict::Array.of(Contributor).meta(omittable: true)
       attribute :event, Types::Strict::Array.of(Event).meta(omittable: true)
       attribute :form, Types::Strict::Array.of(DescriptiveValue).meta(omittable: true)
-      attribute :language, Types::Strict::Array.of(DescriptiveValue).meta(omittable: true)
+      attribute :language, Types::Strict::Array.of(Language).meta(omittable: true)
```
Comment (ndushay) [inline lib/cocina/models/description.rb:10]: not backwards compatible - be prepared for problematic data?

**Comment by ndushay** [comment]
failing tests due to whitespace inconsistencies:

File: lib/cocina/models/descriptive_admin_metadata.rb (hunk 1 above)
Comment (jcoyne) [reply lib/cocina/models/descriptive_admin_metadata.rb:8]: we aren't using this property yet, so probably NBD.

File: lib/cocina/models/descriptive_basic_value.rb (hunk 2 above)
Comment (jcoyne) [reply lib/cocina/models/descriptive_basic_value.rb:17]: we aren't using this yet, NBD.

File: spec/cocina/generator/schema_array_spec.rb [hunk 8]
```diff
@@ -5,14 +5,14 @@
...
-    end
-  end
+  # context 'when an array of datatypes' do
+  #   # DescriptiveBasicValue.standard is an array of strings
+  #   let(:value) { Cocina::Models::DescriptiveBasicValue.new(standard: %w[marc bibframe]) }
+  #
+  #   it 'maps datatypes' do
+  #     expect(value.standard).to eq(%w[marc bibframe])
+  #   end
+  # end
```
Comment (ndushay) [inline spec/cocina/generator/schema_array_spec.rb:15]: this is commented out ... should it be replaced with something else?

Comment (arcadiafalcone) [reply spec/cocina/generator/schema_array_spec.rb:15]: It should probably be deleted -- there are no longer any string arrays in the Description model.

**Comment by jcoyne** [comment]
When I run `exe/generator generate_schema DescriptiveBasicValue` I get an error like this:
```
Unexpected fields for #/components/schemas/DescriptiveBasicValue/allOf/2/properties/standard: description (Openapi3Parser::Error::UnexpectedFields)
```

**Pull request merged**